bookmarklet.js

```diff
@@ -144,5 +160,22 @@ void(
       win.print();
       win.close();
     });
-  })()
-);
+  } else if (sisMatch) {
+    // if the url is using the alternate SIS ID format, get the standard url 
+    // and load it, while prompting the user to run the bookmarklet again after 
+    // the page loads.
+    alert(
+      "It looks like the URL for your Canvas page isn't quite what this tool requires. Your page should now reload with a URL that the tool can use. Once it loads back up, run this bookmark again to produce your syllabus."
+    );
+    const fetchCourse = fetch(`/api/v1/courses/${sisMatch[1]}`)
+      .then((res) => res.json())
+      .then((res) => {
+        window.location.href = `/courses/${res.id}`;
+      });
```

Comment:
FWIW the REST API also supports those `sis_course_id:123456`-style references. So I think you could just check for `match`, then set `match` = `sisMatch` if `match` doesn't exist, and use that in the same URL interpolations. I'm not sure if there's any other reason the script needs the original Canvas course ID vs the `sis_course_id` string, though. If it's only API calls, we probably don't need to redirect at all, just use that string as the course ID.

i.e. in postman, you can query `/api/v1/courses/sis_course_id:281762/activity_stream` or other course-related endpoints and get the same results as if you used the canvas ID.

Reply:
I did notice that, but unfortunately the `sis_course_id` syntax doesn't work as a parameter for the calendar endpoint: `/api/v1/calendar_events?context_codes[]=course_${courseId}&all_events=true` won't work if `courseId` is in the sis format. I didn't see a way to use the SIS ID as a context code, and didn't want to get too far into making a seamless solution for this edge case, so a redirect is where I landed.

Reply:
Cool! I didn't know that, gtk. Thanks!